pkg/storegateway/bucket.go

```diff
@@ -985,15 +902,18 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 		for set.Next() {
 			var series storepb.Series
 
-			stats.mergedSeriesCount++
+			stats.update(func(stats *queryStats) {
```

Comment:
I think we shouldn't call `update()` for each series (it could be many). I would rather use a local unsafe query stats, and then call the `stats.merge()` in the defer defined inside this `tracing.DoWithSpan()`. Makes sense?

_Same comment below when updating `stats.mergedChunksCount`._